R/beamer_presentation.R

```diff
@@ -79,8 +79,11 @@ beamer_presentation <- function(toc = FALSE,
   args <- c()
 
   # template path and assets
-  if (!is.null(template) && !identical(template, "default"))
+  if (!is.null(template)) {
+    if (identical(template, "default"))
+      template <- rmarkdown_system_file("rmd/beamer/default.beamer")
     args <- c(args, "--template", pandoc_path_arg(template))
+  }
```

Comment:
The problem with having our own copy of the default template is getting out of sync with the built-in pandoc templates (e.g .when pandoc starts emitting LaTeX that requires new packages that are included in templates updated along with pandoc). See this commit: https://github.com/rstudio/rmarkdown/commit/82c1dd98c6125f740fd7e2ae57bbaad7c75a10d4

Possibly forking the latest Beamer template gives us the most protection but it doesn't make things 100% future proof.